### ISSUE TITLE
Attendee - Onboarding - Upload a Profile Picture

### ISSUE THREAD

**Issue by Mcscott142** [revelrylabs/bn-mobile-react#419]
_depends on #408_

## Background
A user needs the ability to add a Profile Picture. They can do this upon signing up for the App or later via their Account page. 

See Onboarding flow here: https://www.dropbox.com/s/jiflvjizeywykd5/Sign_in_28_11_2018_only_popup.mp4?dl=0

## Scenario: A user signs up for Big Neon
Given I am signing up for Big Neon
- [ ] When I get to the second screen of Onboarding with name fields
- [ ] Then I am able to upload a profile picture (see video link in the background)

**Comment by brittanygay** [comment]
Pulling this in to start stubbing out the button

**Comment by brittanygay** [comment]
PR is approved for the stubbed out button. Putting this back into the backlog until a developer can pick it up.

**Comment by jwietelmann** [comment]
I made some changes to @brittanygay's branch that should unblock her and also help prepare us to be finished with #418 as well.

**Comment by brittanygay** [comment]
Pulling this back in to finish up

**Comment by brittanygay** [comment]
I just commented with updates here: https://github.com/revelrylabs/bn-mobile-react/pull/432#issuecomment-452861343

I need @jwietelmanns help with two more items, but otherwise I will adjust the design again if we can't set that up right now

**Comment by Mcscott142** [comment]
Getting error when going through onboarding. cc @jwietelmann @daybreaker 

Reproductions Steps:
1) Open App
2) Select `Get Started`
3) Enter Email and Pass
4) Select `Let's do This`
5) Error

```
11:01:22 AM: Unhandled promise rejection, [TypeError: undefined is not an object (evaluating 'data[key].trim')]
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:152:6 in _callTimer
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:200:17 in _callImmediatesPass
- ... 6 more stack frames from framework internals
```

Moving issue back.

**Comment by daybreaker** [comment]
Looks like @brittanygay didnt have merge issues, and this is just some weird bug that pops up when the onboarding code is merged into develop. :-/

**Comment by jwietelmann** [comment]
@Mcscott142 I can't reproduce this?

**Comment by Mcscott142** [comment]
removed node_modules and re-installed, still getting error. Not sure what's going on. Will not let this stop us from at least a stage deploy since it isn't happening for everyone.

**Comment by Mcscott142** [comment]
Will test after the stage deploy I do here shortly.

**Comment by Mcscott142** [comment]
This is still a problem after deploy, turns out not a local issue. 

Via @zmelamed in Slack. Confirmed this morning. 
> It's silently failing on attempting to create an account, can't get past the Lets do this button (login seems to work though)

>Reproductions Steps:
1) Open App
2) Select `Get Started`
3) Enter Email and Pass
4) Select `Let's do This`
5) Error

```
11:01:22 AM: Unhandled promise rejection, [TypeError: undefined is not an object (evaluating 'data[key].trim')]
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:152:6 in _callTimer
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:200:17 in _callImmediatesPass
- ... 6 more stack frames from framework internals
```

cc @jwietelmann @daybreaker

**Comment by Krakaw** [comment]
@Mcscott142 @jwietelmann @daybreaker https://github.com/big-neon/bn-api-node/blob/master/CHANGELOG.md#102-2019-01-16

**Comment by Krakaw** [comment]
Sorry, I just pushed that out ASAP, the explanation is, in the bn-api-node auth / user create pre-processor I trim the keys (except password) but if the field is undefined they would get that error. 
v1.0.2 fixes that bug.

**Comment by Mcscott142** [comment]
Issue is resolved. Moving to UAT.
